Userland/Libraries/LibWeb/HTML/MessageEvent.cpp

```diff
@@ -52,4 +54,17 @@ Variant<JS::Handle<WindowProxy>, JS::Handle<MessagePort>, Empty> MessageEvent::s
     return m_source.value().downcast<JS::Handle<WindowProxy>, JS::Handle<MessagePort>>();
 }
 
+JS::NonnullGCPtr<JS::Object> MessageEvent::ports() const
+{
+    if (!m_ports_array) {
```

Comment:
Is this meant to be SameObject? i.e. does event.ports === event.ports in other engines?

Reply:
In Safari, if I change the test to do this:

```js
let channel = new MessageChannel();
let channel2 = new MessageChannel();
window.postMessage({foo: channel.port1, bar: channel2.port1}, "*", [channel.port1, channel2.port1]);
```

and print out messageEvent.ports === messageEvent.ports, it prints out this:

```
Message 8 data: [object Object]
Message 8 origin: null
Message 8 lastEventId: 
Message 8 source: [object Window]
Message 8 ports: [object MessagePort],[object MessagePort]
Message 8 ports === ports: true
Message 8 source === window: true
Message 8 source === iframe.contentWindow: false
Message 8 source === blobIframe.contentWindow: false
```

Reply:
It's supposed to be a FrozenArray, which also means SameObject in practice:

<https://webidl.spec.whatwg.org/#dfn-create-frozen-array>

> The result of [converting](https://webidl.spec.whatwg.org/#dfn-convert-idl-to-ecmascript-value) an IDL [FrozenArray<T>](https://webidl.spec.whatwg.org/#idl-frozen-array) value to an ECMAScript value is the Object value that represents a reference to the same object that the IDL [FrozenArray<T>](https://webidl.spec.whatwg.org/#idl-frozen-array) represents.

Reply:
Ah fair. That makes me think it would be a good idea to add an `Object.isFrozen(event.ports) === true` test